src/classes/luminator/index.ts

```diff
@@ -0,0 +1,203 @@
+import axios, { AxiosInstance, AxiosPromise, AxiosRequestConfig } from 'axios';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+import { HttpProxyAgent } from 'http-proxy-agent';
+import { config } from '../../config';
+import { replacer } from '../../utils/replacer';
+import {
+  ELuminatiCountry,
+  EStrategyMode,
+  IChangeIp,
+  IConfig,
+  ICreateProxy,
+  ICreateProxyAgents,
+  ILuminatiConfig,
+  TStrategy,
+} from './types';
+
+/**
+ * @description Luminator class.
+ */
+export class Luminator {
+  public axios: AxiosInstance;
+  public sessionId: number;
+  public country: ELuminatiCountry;
+
+  private readonly luminatiConfig: ILuminatiConfig;
+  private readonly strategy: TStrategy;
+
+  /**
+   * @constructor
+   * @param {IConfig} config
+   */
+  constructor(config: IConfig) {
+    this.luminatiConfig = config.luminatiConfig;
+    this.strategy = config.strategy;
+
+    // Throw an error is country array is empty
+    if (this.strategy && this.strategy.mode === EStrategyMode.CHANGE_IP_EVERY_REQUESTS) {
+      Luminator.checkIfCountriesArrayIsntEmpty(this.strategy.countries);
+    }
+
+    if (config.axiosConfig) {
+      this.axios = axios.create({ ...config.axiosConfig, proxy: false });
+    } else {
+      this.axios = axios.create({ proxy: false });
+    }
+  }
+
+  /**
+   * @description Generate a new agent.
+   * @param {IChangeIp} [params] - Params to handle multiple strategies.
+   * @returns {Luminator}
+   */
+  changeIp(params?: IChangeIp): Luminator {
+    // Creates an agent with a random countries and sessionId
+    if (!params) {
+      const { httpAgent, httpsAgent }: ICreateProxyAgents = this.createProxyAgents({
+        country: this.getRandomCountry(),
+        sessionId: Luminator.randomNumber(config.session.randomLimit.min, config.session.randomLimit.max),
+      });
+
+      this.axios.defaults.httpsAgent = httpsAgent;
+      this.axios.defaults.httpAgent = httpAgent;
+
+      return this;
+    }
+
+    // Creates an agent with specific countries and a specific sessionId
+    if (params.countries && params.sessionId) {
+      Luminator.checkIfCountriesArrayIsntEmpty(params.countries);
+
+      const { httpAgent, httpsAgent }: ICreateProxyAgents = this.createProxyAgents({
```

Comment:
the method should be called "setIp". This is more generic :  
According to the parameter it could change countries & ip, or just keep the session passed in parameter. : )

Reply:
- Nothing can guarantee that a same country & session will provide the same IP. I think that's a "best effort" feature, not more. Especially with a residential area.

- I'll offer the possibility to set country without session (and vice versa) because because I think that one day this one will be useful. (But I agree, this one isn't useful for us for the moment)

- `changeIp` => can be renamed in `setIp`

Reply:
OK